Remove redundant error messages

hqhq opened the pull request:
For operations on multi containers, we printed error for each
failed container, then printed an extra message for container
names, it seems redundant.

Addresses comments:
https://github.com/docker/docker/pull/15078#discussion_r47988449

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

hqhq commented:
ping @duglin

duglin commented on api/client/kill.go:
```diff
@@ -17,17 +18,16 @@ func (cli *DockerCli) CmdKill(args ...string) error {
+	var errs []string
 	for _, name := range cmd.Args() {
 		if err := cli.client.ContainerKill(name, *signal); err != nil {
-			fmt.Fprintf(cli.err, "%s\n", err)
```
I think it might be best to keep this fmt.Fprintf() here so that we show the status for each container in the other in which the user provided them. Then remove the "if' down below on line 29. Then you can get rid of "errs" too.
Shouldn't this PR wait for the "set" one to be merged first?

duglin replied:
Oh I see, you changed "set".  You could do it that way but if people dislike this PR then you'll have to "fix" set().  Guess you're being optimistic. :-)

hqhq replied:
@duglin The problem here is we have to return `error` in `CmdKill`, if we `fmt.Fprintf` here, what do you expect for the return value? I don't think it's a good idea to change return value types here.
And I don't think we have an order issue, slices will keep the order as we appended, no?

Yeah I'm being optimistic since I don't see much options here, but I'm always willing and ready for comments and modifications. :smile:

calavera commented:
LGTM

thaJeztah commented on api/client/rmi.go:
```diff
@@ -38,8 +39,7 @@ func (cli *DockerCli) CmdRmi(args ...string) error {
 		if err != nil {
-			fmt.Fprintf(cli.err, "%s\n", err)
-			errNames = append(errNames, name)
+			errs = append(errs, fmt.Sprintf("Failed to remove images (%s): %s", name, err))
```
Should this be singular? `Failed to remove image`

duglin commented:
I like the idea of removing the duplicate error message, however, I'm worried about the UX change and would like to know what others think. Right now, as each arg is processed we print something to the screen - either name/id of the resource in the happy case or the error message. The point is that there's a consistency between the order of the args on the cmd line and the order of output text (ignoring the stdout vs stderr difference).  With this PR that changes. Now users will see the list of ids/names that worked and then the list of ids/names that failed - which will most likely not be the same order as the cmd line.  This means that if people have any kind of expectations of ordering then we're going to break them.

I have no idea if this is an issue for people or not, nor am I sure how we can know for sure. 

Just guessing, but I suspect this won't be an issue but I wanted to bring it up so people could comment and think about it.

/ping @docker/maintainers

duglin commented:
LGTM once @thaJeztah's comment is addressed.

Despite this being the 2nd LGTM I'd prefer to wait for more due to the UX impact I mentioned in https://github.com/docker/docker/pull/18768#issuecomment-167784277

vdemeester commented:
```bash
$ docker kill 9e 1b
Failed to kill container (9e): Error response from daemon: Cannot kill container 9e: nosuchcontainer: No such container: 9e
Failed to kill container (1b): Error response from daemon: Cannot kill container 1b: nosuchcontainer: No such container: 1b
$ docker rm 83
Failed to remove container (83): Error response from daemon: Conflict, You cannot remove a running container. Stop the container before attempting removal or use -f
```

But it looks like it's on the server side (all the repetitions). So, LGTM for me :wink:.

hqhq committed:
Remove redundant error messages

For operations on multi containers, we printed error for each
failed container, then printed an extra message for container
names, it seems redundant.

Addresses comments:
https://github.com/docker/docker/pull/15078#discussion_r47988449

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@thaJeztah @duglin Updated, and added change for `api/client/update.go` as well.

thaJeztah commented:
Thanks!

duglin commented:
during the maintainer review session today we decided we're ok with the reordering of the output.
So we can merge this now.
thanks! @hqhq
